### PR TITLE
Add EditAppointmentCommandParser Tests

### PR DESCRIPTION
**Pull request by Fuzanwenn**

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/EditAppointmentCommandParserTest.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    public void parse_oneFieldSpecified_success() {
+        // name
+        Index targetIndex = INDEX_THIRD_APPOINTMENT;
+        String userInput = targetIndex.getOneBased() + NAME_DESC_AMY;
+        EditAppointmentCommand.EditAppointmentDescriptor descriptor =
+                new EditAppointmentDescriptorBuilder().withName(VALID_NAME_AMY).build();
+        EditAppointmentCommand expectedCommand = new EditAppointmentCommand(targetIndex, descriptor);
+        assertParseSuccess(parser, userInput, expectedCommand);
+
+        // phone
```
Comment (TZL0) [inline src/test/java/seedu/address/logic/parser/EditAppointmentCommandParserTest.java:146]: Is it possible to remove this part?

File: src/test/java/seedu/address/logic/parser/EditAppointmentCommandParserTest.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        assertParseSuccess(parser, userInput, expectedCommand);
+
+        // address
+        userInput = targetIndex.getOneBased() + MEDICAL_TEST_DESC_7;
+        descriptor = new EditAppointmentDescriptorBuilder().withMedicalTest(VALID_MEDICAL_TEST_7).build();
+        expectedCommand = new EditAppointmentCommand(targetIndex, descriptor);
+        assertParseSuccess(parser, userInput, expectedCommand);
+
+        // tags
+        userInput = targetIndex.getOneBased() + SLOT_DESC_7;
```
Comment (TZL0) [inline src/test/java/seedu/address/logic/parser/EditAppointmentCommandParserTest.java:165]: Is it slot?

File: src/test/java/seedu/address/logic/parser/EditAppointmentCommandParserTest.java [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        expectedCommand = new EditAppointmentCommand(targetIndex, descriptor);
+        assertParseSuccess(parser, userInput, expectedCommand);
+
+        // email
+        userInput = targetIndex.getOneBased() + DOCTOR_DESC_7;
+        descriptor = new EditAppointmentDescriptorBuilder().withDoctor(VALID_DOCTOR_CAITIE).build();
+        expectedCommand = new EditAppointmentCommand(targetIndex, descriptor);
+        assertParseSuccess(parser, userInput, expectedCommand);
+
+        // address
```
Comment (TZL0) [inline src/test/java/seedu/address/logic/parser/EditAppointmentCommandParserTest.java:158]: Is it medical test?

File: src/test/java/seedu/address/logic/parser/EditAppointmentCommandParserTest.java [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        EditAppointmentCommand expectedCommand = new EditAppointmentCommand(targetIndex, descriptor);
+        assertParseSuccess(parser, userInput, expectedCommand);
+
+        // phone
+        userInput = targetIndex.getOneBased() + NAME_DESC_AMY;
+        descriptor = new EditAppointmentDescriptorBuilder().withName(VALID_NAME_AMY).build();
+        expectedCommand = new EditAppointmentCommand(targetIndex, descriptor);
+        assertParseSuccess(parser, userInput, expectedCommand);
+
+        // email
```
Comment (TZL0) [inline src/test/java/seedu/address/logic/parser/EditAppointmentCommandParserTest.java:152]: Is it doctor?

File: src/test/java/seedu/address/testutil/EditAppointmentDescriptorBuilder.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    public EditAppointmentDescriptorBuilder() {
+        descriptor = new EditAppointmentCommand.EditAppointmentDescriptor();
+    }
+
+    public EditAppointmentDescriptorBuilder(EditAppointmentCommand.EditAppointmentDescriptor descriptor) {
+        this.descriptor = new EditAppointmentCommand.EditAppointmentDescriptor(descriptor);
+    }
+
+    /**
+     * Returns an {@code EditPersonDescriptor} with fields containing {@code person}'s details
```
Comment (TZL0) [inline src/test/java/seedu/address/testutil/EditAppointmentDescriptorBuilder.java:26]: EditAppointmentDescriptor?

**Review by TZL0 — CHANGES_REQUESTED** [review]

**Review by TZL0 — APPROVED** [review]

**Review by SpecOps2016 — APPROVED** [review]
LGTM

**Pull request merged**